### PR TITLE
Sugar to builtins, update list lib, prevent overwrite

### PR DESCRIPTION
**Pull request by akalicki**
- change syntactic sugar to real operators equating to builtin functions
- add `_` to builtin funcs to prevent user overwrite and breaking of corresponding operators
- updated list library to use new `::` cons operator

### CODE REVIEW

File: test/compiler/pass/_list_ops.ods [hunk 1]
```diff
@@ -15,8 +15,5 @@ do print(len([]))
 do print(len(ints))
 
 /* list cons test */
-do z = cons(5, ints)
```
Comment (alexandramedway) [inline test/compiler/pass/_list_ops.ods]: we're getting rid of this bc we assume if :: is broken then this is too?

**Comment by alexandramedway** [comment]
Looks good to me!

**Comment by alexandramedway** [comment]
tumblr_lqaynotksh1qggk3do1_500

**Comment by dannyech** [comment]
@akalicki make sure to update the stdlib part of the Final Report. Otherwise looks gouchy to me

File: test/compiler/pass/_list_ops.ods (hunk 1 above)
Comment (akalicki) [inline test/compiler/pass/_list_ops.ods]: Now you have to use `::` to prepend to a list. `::` corresponds to the builtin function `_cons`, which can't be overwritten or called by the user.

Comment (alexandramedway) [inline test/compiler/pass/_list_ops.ods]: Ah. I like this much better

**Comment by alexandramedway** [comment]
tumblr_lyr2bl4oeo1r8lg7to1_400

**Comment by lillyfwang** [comment]
!(Ryan love)[http://41.media.tumblr.com/tumblr_lyr2bl4OEO1r8lg7to1_400.jpg]

**Comment by lillyfwang** [comment]
Lolol @alexandramedway great minds think alike. Good thing your comment didn't fail the way mine did.

**Pull request merged**